Zend/zend_smart_str.c

```diff
@@ -169,3 +169,30 @@ ZEND_API void ZEND_FASTCALL _smart_string_alloc(smart_string *str, size_t len)
 		str->c = erealloc2(str->c, str->a + 1, str->len);
 	}
 }
+
+ZEND_API void ZEND_FASTCALL smart_str_append_truncated(smart_str *str, zend_string *value, size_t length)
```

Comment:
If we want to export this API (do we?), I think it should be `smart_str_append_escaped_truncated` ... from the current name, I would have assumed that this just performs a simple truncation.

Reply:
I think we do want to export it, it may be helpful for extensions. The name is silly long now ...